frontend/packages/dev-console/integration-tests/support/step-definitions/addFlow/page-details.ts

```diff
@@ -103,39 +103,46 @@ Then('user will see {string} option', (addPageOption: string) => {
   verifyAddPage.verifyAddPageCard(addPageOption);
 });
 
-When('user enable Details toggle', () => {
-  if (cy.get(addPagePO.detailsOnOffText).contains('Details on')) {
-    cy.log('Details are on');
-  } else {
-    cy.get(addPagePO.detailsOnOffSwitch).click();
-  }
+Given('add page Details toggle shows {string}', (label: string) => {
+  // Just waiting until the switch with all options is shown.
+  // This doesn't ensure that the right label is shown because its hidden via CSS!
+  cy.get(addPagePO.detailsOnOffSwitch)
+    .find('.pf-c-switch__label')
+    .should('contain', label);
```

Comment:
The problem here is that both labels are "displayed" in the dom.

It is a CSS rule that shows or hides the on or the off label based on an invisible input checkbox.

Also `be.visible` doesn't check if the element is _really_ visible to the user. The CSS rule adds `display: none` to one of both spans.

But for that reason, I wanted to check if the label "exists" (via `contains`) and don't want to pretend to anyone that this checks if the label is visible or not. If you prefer I can remove this check at all because of this confusion.